graph/skills/index.py

```
@@ -0,0 +1,268 @@
+        bak_path = self._db_path + ".bak"
+        if os.path.exists(self._db_path):
+            try:
+                shutil.copy2(self._db_path, bak_path)
+                os.remove(self._db_path)
+                log.warning(
+                    "[skills] incompatible schema — backed up %s → %s and will rebuild",
+                    self._db_path,
+                    bak_path,
+                )
+            except OSError as exc:
+                log.error("[skills] backup failed: %s — will attempt in-place schema reset", exc)
```

Ruff (0.15.10)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)